### PR TITLE
feat: Allow Multiple Attributes

### PR DESCRIPTION
**Pull request by trenaryja**
This resolves issue #274

### CODE REVIEW

File: next-themes/src/types.ts [hunk 1]
```diff
@@ -34,8 +36,8 @@ export interface ThemeProviderProps {
   storageKey?: string | undefined
   /** Default theme name (for v0.0.12 and lower the default was light). If `enableSystem` is false, the default theme is light */
   defaultTheme?: string | undefined
-  /** HTML attribute modified based on the active theme. Accepts `class` and `data-*` (meaning any data attribute, `data-mode`, `data-color`, etc.) */
-  attribute?: `data-${string}` | 'class' | undefined
+  /** HTML attribute modified based on the active theme. Accepts `class`, `data-*` (meaning any data attribute, `data-mode`, `data-color`, etc.), or an array which could include both */
+  attribute?: `data-${string}` | 'class' | Attribute[] | undefined
```
Comment (pacocoursey) [inline next-themes/src/types.ts:40]:
```suggestion
  attribute?: Attribute | Attribute[] | undefined
```

File: next-themes/src/index.tsx [hunk 2]
```diff
@@ -52,18 +52,22 @@ const Theme = ({
...
-    } else {
-      if (name) {
-        d.setAttribute(attribute, name)
-      } else {
-        d.removeAttribute(attribute)
+    const handleAttribute = (attr: Attribute) => {
+      if (attr === 'class') {
+        d.classList.remove(...attrs)
+        if (name) d.classList.add(name)
+      } else if (attr.includes('data-')) {
```
Comment (pacocoursey) [inline next-themes/src/index.tsx:59]:
```suggestion
      } else if (attr.startsWith('data-')) {
```

**Review by pacocoursey — APPROVED** [review]

**Comment by pacocoursey** [comment]
Mind reviewing my comments and the small conflict? Will merge after that.

**Comment by trenaryja** [comment]
> Mind reviewing my comments and the small conflict? Will merge after that.

No problem, especially when you practically did the work lol. Committed the updates, thanks for taking a look!

**Pull request merged**